database/database/connection.py

```diff
@@ -515,3 +515,552 @@ def get_latest_products_with_unknown_sustainability_label(self) -> pd.DataFrame:
+                            + get_max_sustainability_scores.c.eco_quality
+                            + get_max_sustainability_scores.c.eco_energy
+                            + get_max_sustainability_scores.c.eco_waste_air
+                            + get_max_sustainability_scores.c.eco_environmental_management
+                        )
+                        / 8
+                    ).label("ecological_score"),
+                    (
+                        func.sum(
+                            get_max_sustainability_scores.c.social_labour_rights
+                            + get_max_sustainability_scores.c.social_business_practice
+                            + get_max_sustainability_scores.c.social_social_rights
+                            + get_max_sustainability_scores.c.social_company_responsibility
+                            + get_max_sustainability_scores.c.social_conflict_minerals
+                        )
+                        / 5
+                    ).label("social_score"),
+                    get_max_sustainability_scores.c.mean_credibility,
+                )
+                .group_by(
+                    get_max_sustainability_scores.c.prod_id,
+                    get_max_sustainability_scores.c.mean_credibility,
+                )
+                .subquery()
+            )
+            return (
+                db_session.query(
+                    get_eco_and_social_means.c.prod_id,
+                    get_eco_and_social_means.c.mean_credibility,
+                    get_eco_and_social_means.c.ecological_score,
+                    get_eco_and_social_means.c.social_score,
+                    func.round(
+                        (
+                            func.sum(
+                                get_eco_and_social_means.c.ecological_score
+                                + get_eco_and_social_means.c.social_score
+                            )
+                            / 2
+                        ),
+                        0,
+                    ).label("sustainability_score"),
+                )
+                .group_by(
+                    get_eco_and_social_means.c.prod_id,
+                    get_eco_and_social_means.c.mean_credibility,
+                    get_eco_and_social_means.c.ecological_score,
+                    get_eco_and_social_means.c.social_score,
+                )
+                .subquery()
+            )
+
+    def get_rank_by_sustainability(self, agregated_by: str) -> pd.DataFrame:
+        """
+        This function ranks unique credible products by its aggregated credibility.
+
+        Args:
+            aggregated_by (str): Determines product attribute to aggregated data.
+
+        Returns:
+            pd.DataFrame: Query results as `pd.Dataframe`.
+        """
+        with self._session_factory() as db_session:
+            unique_credible_products = self.calculate_sustainability_scores()
+
+            if agregated_by == "merchant":
```

Comment:
This conditional branch could be simplified by defining a map with the following format:
```
aggregation_map = {
"merchant": self._database_class.merchant,
"category": self._database_class.category,
"brand": self._database_class.brand
}
```

And then just call:
```
                    db_session.query(
                        aggregation_map[aggregated_by]),
                        func.round(func.avg(unique_credible_products.c.sustainability_score)).label(
                            "sustainability_score"
                        ),
                    )
                    .join(
                        unique_credible_products,
                        unique_credible_products.c.prod_id == self._database_class.id,
                    )
                    .group_by(aggregation_map[aggregated_by])
                    .order_by(desc("sustainability_score"))
                    .subquery()
                )
```
It will be a lot cleaner and reduce redundancy

Reply:
Fixed in [ef07613](https://github.com/calgo-lab/green-db/pull/103/commits/ef07613721d55dc5de3159512bcedbe5afd3b8cf)